domain/schema/controller.go

```diff
@@ -482,3 +485,53 @@ BEGIN
     VALUES (4, 5, OLD.key, DATETIME('now'));
 END;`)
 }
+
+func upgradeInfoSchema() database.Delta {
+	return database.MakeDelta(`
+CREATE TABLE upgrade_info (
+    uuid             TEXT PRIMARY KEY,
+    previous_version TEXT NOT NULL,
+    target_version   TEXT NOT NULL,
+    created_at       TIMESTAMP NOT NULL,
+    started_at       TIMESTAMP,
+    completed_at     TIMESTAMP
+);
+
+CREATE TABLE upgrade_info_controller_node (
+    uuid                      TEXT PRIMARY KEY,
+    controller_node_id        TEXT NOT NULL,
+    upgrade_info_uuid         TEXT NOT NULL,
+    node_upgrade_completed_at TIMESTAMP,
+    CONSTRAINT                fk_controller_node_id
```

Comment:
You can't actually do this check here. It will need to be verified by a join to controller node at the time of record creation.

Consider the scenario where the controller plane changes, but you can't remove a node because there is a FK relationship with an upgrade record.